### DISCUSSION TITLE
Z8102AX-T Compatibility

### DISCUSSION THREAD

**Discussion by Code404404** [4IceG/ZBT-Z8102AX-M2#3] [Q&A]
Hello, can I install this firmware into my ZBT Z8102AX-T? My Z8102AX-T uses factory firmware which is Z8102AX-nand-mt7981-DDR4-23.0804_121617.

**Comment by 4IceG** [comment]
Hi @Code404404 ,
The M2 and T versions differ slightly even though they have the same motherboard version. 

If the manufacturer has not changed anything, the firmware should be compatible.

**Reply by Code404404** [reply]
If I install it, can it bricked? I able to use firmware from http://www.aturnofthenut.com/autobuilds/ which this firmware is compatible for both versions.

**Reply by 4IceG** [reply]
If you installed ROOter software, mine should also work properly.

**Reply by Code404404** [reply]
Which file should I use? kernel, factory or sysupgrade? Because ROOter firmware only have 1 file.

**Reply by 4IceG** [reply]
The only question is what modem do you have in the router, because my fw may not support all available modems.

**Reply by Code404404** [reply]
Quectel RM502Q-AE

**Reply by 4IceG** [reply]
First install fw ROOter and then update with the sysupgrade file without saving the configuration.

**Reply by Code404404** [reply]
Now Im using stock firmware Z8102AX-nand-mt7981-DDR4-23.0804_121617. Which file I should use?

**Reply by 4IceG** [reply]
First install firmware from http://www.aturnofthenut.com/autobuilds

there should also be installation instructions.

**Reply by Code404404** [reply]
I managed to install it but cannot connect to the internet. It detected my modem but not showing up in the information page. I don know what happening to the modem because it doesn't have log info like ROOter. Some AT commands I can send and manage to lock the band.

**Reply by 4IceG** [reply]
My fw is usually Openwrt with a few additional packages. You need to configure the modem normally and the qmi/mbim protocol. You should have at commands to set your modem to mbim/qmi mode.

**Reply by Code404404** [reply]
Still not working after changing between QMI and MBIM protocol.

**Reply by 4IceG** [reply]
I hope you remembered to add the WAN interface to the firewall. If you did this when creating the mbim/qmi protocols, I don't have any other ideas, because there shouldn't be any more settings to do here. I am leaving out cases when the mobile network operator has some strange requirements.

**Reply by 4IceG** [reply]
The procedure is as follows: Set the modem as mbim/qmi. Create a new wan interface with the qmi/mbim protocol. Add wan to the firewall and it must work.
